packages/discord.js/src/structures/CommandInteractionOptionResolver.js

```diff
@@ -142,12 +142,24 @@ class CommandInteractionOptionResolver {
    * Gets a channel option.
    * @param {string} name The name of the option.
    * @param {boolean} [required=false] Whether to throw an error if the option is not found.
+   * @param {ChannelType[]} [channelTypes=[]] The allowed types of channels. If empty, all channel types are allowed.
    * @returns {?(GuildChannel|ThreadChannel|APIChannel)}
    * The value of the option, or null if not set and not required.
    */
-  getChannel(name, required = false) {
+  getChannel(name, required = false, channelTypes = []) {
```

Comment:
Thoughts on the second parameter being an object instead?

Reply:
Breaking though

Reply:
We can implement it in a non-breaking way with a deprecation message emitting if one doesn't use an object... though this introduces some inconsistency in the second parameter of other methods.